addons/mail/static/tests/new/chat_window/chat_window_manager_tests.js

```diff
@@ -0,0 +1,99 @@
+
+let target;
+
+QUnit.module("chat window manager", {
+    async beforeEach() {
+        target = getFixture();
+    },
+});
+
+QUnit.test("chat window does not fetch messages if hidden", async function (assert) {
+    const pyEnv = await startServer();
+    pyEnv["mail.channel"].create([
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+    ]);
+    patchUiSize({ width: 900 });
+    assert.ok(
+        CHAT_WINDOW_END_GAP_WIDTH * 2 + CHAT_WINDOW_WIDTH * 2 + CHAT_WINDOW_INBETWEEN_WIDTH < 900
+    );
+    assert.ok(
+        CHAT_WINDOW_END_GAP_WIDTH * 2 + CHAT_WINDOW_WIDTH * 3 + CHAT_WINDOW_INBETWEEN_WIDTH * 2 >
+            900
+    );
+    await start({
+        mockRPC(route, args) {
+            if (route === "/mail/channel/messages") {
+                assert.step("fetch_messages");
+            }
+        },
+    });
+    assert.containsN(target, ".o-mail-chat-window", 2);
+    assert.containsOnce(target, ".o-mail-chat-window-hidden-menu");
+    assert.verifySteps(["fetch_messages", "fetch_messages"]);
+});
+
+QUnit.test("click on hidden chat window should fetch its messages", async function (assert) {
+    const pyEnv = await startServer();
+    pyEnv["mail.channel"].create([
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+        {
+            channel_member_ids: [
+                [0, 0, { is_minimized: true, partner_id: pyEnv.currentPartnerId }],
+            ],
+        },
+    ]);
+    patchUiSize({ width: 900 });
+    assert.ok(
+        CHAT_WINDOW_END_GAP_WIDTH * 2 + CHAT_WINDOW_WIDTH * 2 + CHAT_WINDOW_INBETWEEN_WIDTH < 900
+    );
+    assert.ok(
+        CHAT_WINDOW_END_GAP_WIDTH * 2 + CHAT_WINDOW_WIDTH * 3 + CHAT_WINDOW_INBETWEEN_WIDTH * 2 >
+            900
+    );
+    await start({
+        mockRPC(route, args) {
+            if (route === "/mail/channel/messages") {
+                assert.step("fetch_messages");
+            }
+        },
+    });
+    assert.containsN(target, ".o-mail-chat-window", 2);
+    assert.containsOnce(target, ".o-mail-chat-window-hidden-menu");
+    assert.verifySteps(["fetch_messages", "fetch_messages"]);
+    await click(".o-mail-chat-window-hidden-menu");
+    await click(".o-mail-chat-window-hidden-menu-item .o-mail-chat-window-header");
+    assert.verifySteps(["fetch_messages"]);
```

Comment:
It's easier as 2 different `verifySteps()`: we clearly relate the steps to when they happen. Here it's clear the new `fetch_messages` come from showing the hidden chat window